src/icmp/Icmp4.cc

```diff
@@ -111,7 +113,7 @@ Icmp4::SendEcho(Ip::Address &to, int opcode, const char *payload, int len)
     ++icmp_pkts_sent;
 
     // Construct ICMP packet data content
-    echo = (icmpEchoData *) (icmp + 1);
+    echo = (icmpEchoData *) ((char *) icmp + sizeof(*icmp));
```

Comment:
`pkt` is an array of chars so I hope that something like this works:

```suggestion
    echo = reinterpret_cast<icmpEchoData*>(&pkt[sizeof(*icmp)]);
```

but, if necessary, we can be more explicit:

```suggestion
    echo = reinterpret_cast<icmpEchoData*>(reinterpret_cast<char*>(pkt) + sizeof(*icmp));
```